pkg/utils/utils.go

```diff
@@ -158,121 +181,156 @@ func InitInformerMap() {
 
 // GetObjectMetaData returns metadata of the given object
 func GetObjectMetaData(obj interface{}) metaV1.ObjectMeta {
+	metadata := obj.(*unstructured.Unstructured).UnstructuredContent()["metadata"]
+	objectMeta := metaV1.ObjectMeta{}
+	jsonbody, err := json.Marshal(metadata)
+	if err != nil {
+		log.Logger.Infof("Error while marshalling: %v", err)
+	}
+	if err := json.Unmarshal(jsonbody, &objectMeta); err != nil {
+		log.Logger.Info("Error while Unmarshalling: %v", err)
+	}
 
-	var objectMeta metaV1.ObjectMeta
-
-	switch object := obj.(type) {
-	case *coreV1.Event:
-		objectMeta = object.ObjectMeta
-		// pass InvolvedObject`s annotations into Event`s annotations
-		// for filtering event objects based on InvolvedObject`s annotations
+	if GetObjectTypeMetaData(obj).Kind == "Event" {
```

Comment:
We can use `unstructured.GetKind()` here instead of getting `GetObjectTypeMetaData`. We can actually get rid of `GetObjectTypeMetaData` and `GetObjectMetaData` since we can get all the values from methods implemented by unstructured

Reply:
Sure we can @PrasadG193 , but I think we can take it in a refactor PR, rather than taking all the changes in this one.
What's your thought?